aws/data_source_aws_s3_bucket_object.go

```diff
@@ -27,6 +27,10 @@ func dataSourceAwsS3BucketObject() *schema.Resource {
 				Type:     schema.TypeString,
 				Required: true,
 			},
+			"bucket_key_enabled": {
```

Comment:
Oh! While we are adjusting this pull request, we can also queue up the changelog entries by creating a `.changelog/16581.txt` file with contents:

``````markdown
```release-note:enhancement
data-source/aws_s3_bucket_object: Add `bucket_key_enabled` attribute (Support S3 Bucket Keys)
```

```release-note:enhancement
resource/aws_s3_bucket: Add `bucket_key_enabled` argument to `server_side_encryption_configuration` `rule` configuration block (Support S3 Bucket Keys)
```

```release-note:enhancement
resource/aws_s3_bucket_object: Add `bucket_key_enabled` attribute (Support S3 Bucket Keys)
```
``````

Reply:
Nice!